### PR TITLE
Add sort command to ModuleMateFinder

### PR DESCRIPTION
**Pull request by lawwm**
Add sort command to ModuleMateFinder. User guide excerpt for sorting below.

Most fields of Person now implement comparable interface. Add an ``o/`` tag for order. Sort command passes comparator to Addressbook model.

Barebones tests added, will add more tests later.

**UserGuide Excerpt**

Format: `sort [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [s/STATUS] [t/TAG] [o/ORDER]​`

* Sorts list with specified field(s). For any two persons, latter fields will only be considered if preceding fields are equal.​ 
* Order of parameters is important except for that of `o/ORDER`.
* At least one of the optional fields must be provided (excluding order).
* Parameters are ignored except for order `o/asc`.
* Orders are optional and default order is "asc". Parameters of order must be either "asc" or "desc" and is case-insensitive. 

Examples:
* `sort n/ p/`  will sort the list by name first. If two persons have the same name, then sort by phone number.
* `sort n/ o/desc` will sort the list by name in descending order.

### CODE REVIEW

**Review by Btaykb — COMMENTED** [review]
LGTM, nice job implementing the comparable interface for the different fields.

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        public PersonComparator(List<Prefix> fields, String order) {
+            this.fields = fields;
+            this.order = order;
+        }
+
+        @Override
+        public int compare(Person o1, Person o2) {
+            for (Prefix field : fields) {
+                int result = 0;
+                if (PREFIX_NAME.equals(field)) {
```
Comment (bakano98) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:82]: Minor, but may look cleaner if we use `switch case` statements here instead.

**Review by bakano98 — COMMENTED** [review]
I've commented on some minor stuff, but other than that, LGTM!
We'll wait for @fantablack to finish reviewing first before merging.

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 1 above)
Comment (lawwm) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:82]: Cannot use Prefix class for ``switch`` statement.
>The type of the Expression must be char, byte, short, int, Character, Byte, Short, Integer, String, or an enum type (§8.9), or a compile-time error occurs.

https://stackoverflow.com/questions/22185706/using-java-lang-class-in-a-switch-statement

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_STATUS + "STATUS] "
+            + "[" + PREFIX_TAG + "TAG] "
+            + "[" + PREFIX_ORDER + "ORDER]\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_TAG + " " + PREFIX_NAME
+            + " " + PREFIX_ORDER + "desc" + "\n";
+
+    public static final String MESSAGE_SUCCESS = "Sorted Modules successfully";
```
[the remark this answers is not in the document]
Comment (lawwm) [reply src/main/java/seedu/address/logic/commands/SortCommand.java]: Ok, made the necessary changes and committed below.

**Pull request merged**